refactor(NodeService): create NodeInfoService and remove UpgradeModule dependency

breity opened the pull request:
## Changes
- Creates NodeInfoService.
- Moves showNodeInfo method from NodeService to NodeInfoService and removes the UpgradeModule dependency from the NodeService.

Closes #615.

## Test
Make sure you can still view node info dialogs from:
- Grade by Step (NodeGrading)
- Milestone Reports

breity committed:
refactor(NodeService): create NodeInfoService and remove UpgradeModule
#615

hirokiterashima reviewed (changes requested):
Should the NodeInfoService be in MilestoneDetails and NodeGradingView, and not in NodeService?

So MilestoneDetails.showMilestoneStepInfo would call NodeInfoService.showNodeInfo
```
  showMilestoneStepInfo($event) {
    this.NodeInfoService.showNodeInfo(this.milestone.nodeId, $event);
  }
```

and NodeGradingView.showRubric would call NodeInfoService.showNodeInfo
```
  showRubric($event) {
    this.NodeInfoService.showNodeInfo(this.nodeId, $event);
  }
```

This way we can 
1. remove NodeInfoService dependency from NodeService
2. remove NodeService dependency from MilestoneDetails and NodeGradingView

What do you think?

breity committed:
refactor(Node Service): remove NodeInfoService dependency
- Removes NodeService dependency from MilestoneDetails and NodeGradingView
#615

breity commented:
@hirokiterashima Yep, makes sense. Thanks. I updated with those changes.

hirokiterashima committed:
refactor(Node Service): Lower-case variable names #615

hirokiterashima reviewed (approved):
Looks good. Remember to lower-case all new variable names.
